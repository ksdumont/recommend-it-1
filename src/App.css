h1 {
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  font-size: 20px;
  font-weight: bold;
  margin: 20px;
}

h2 {
  font-weight: bold;
  font-size: 25px;
  text-align: center;
}

.button.is-primary {
  background-color: #037766;
}

nav.navbar {
  background-color: #fef2e5;
  margin-bottom: 20px;
}

.tile {
  flex-wrap: wrap;
  justify-content: space-around;
}

.card {
  width: 45%;
  margin: 20px;
  padding-bottom: 50px;
  color: #000;
}

.card.SubReddit {
  background-color: lightpink;
}
.card.Podcast {
  background-color: lightsteelblue;
}
.card.Book {
  background-color: lightgray;
}

.card.Show {
  background-color: lightgreen;
}

.card.Movie {
  background-color: lightskyblue;
}

.card.Youtube.Channel {
  background-color: lightcoral;
}

.card.Website {
  background-color: lightsalmon;
}

.card-footer {
  position: absolute;
  width: 100%;
  bottom: 0;
}

a.card-footer-item {
  text-align: center;
  color: darkblue;
}

.card.following .author-name {
  background-color: lightyellow;
}

p.numberOfFollowers,
p.follower-button {
  text-align: center;
}

form {
  text-align: center;
  margin-bottom: 50px;
}

input,
select {
  text-align: center;
  font-size: 20px;
  padding: 10px;
  width: 80%;
  margin: 10px;
}

.homepage {
  max-width: 90%;
  margin: 0 auto;
}

@media screen and (max-width: 815px) {
  .card {
    width: 90%;
  }
}

@media screen and (max-width: 1087px) {
  .navbar-brand,
  .buttons {
    justify-content: center;
  }

  .navbar-menu {
    display: block;
  }
}
