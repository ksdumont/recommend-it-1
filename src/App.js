import React, { useEffect, useState } from "react";
import "./App.css";
import firebase from "./firebase";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Dashboard from "./components/Dashboard";
import HomePage from "./components/HomePage";
import Login from "./components/Login";
import Register from "./components/Register";
import Edit from "./components/Edit";
import Category from "./components/Category";
import Author from "./components/Author";

function App() {
  const [firebaseInitialized, setFirebaseInitialized] = useState(false);

  useEffect(() => {
    firebase.isInitialized().then(val => {
      setFirebaseInitialized(val);
    });
  });

  return firebaseInitialized !== false ? (
    <section className="app-wrapper">
      <Router>
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/register" component={Register} />
          <Route exact path="/dashboard" component={Dashboard} />
          <Route exact path="/editRecommendation/:id" component={Edit} />
          <Route exact path="/category/:category" component={Category} />
          <Route exact path="/author/:author" component={Author} />
        </Switch>
      </Router>
    </section>
  ) : (
    <header className="app-header" id="header" title="Header">
      <p>Another Paragraph</p>
      <h1>RecommendIt</h1>
      <p className="loading">Loading...</p>
    </header>
  );
}

export default App;
