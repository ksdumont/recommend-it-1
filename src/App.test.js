import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<App /> testing", () => {
  const wrapper = shallow(<App />);

  it("header should have an h1 and p", () => {
    expect(wrapper.find("header").children().length).toBe(3);
  });

  it("should contain one h1", () => {
    expect(wrapper.find("h1").length).toBe(1);
  });
  it("h1 should contain the text RecommendIt", () => {
    expect(wrapper.find("h1").text()).toBe("RecommendIt");
  });

  it("should containe at least 1 p with the class of loading", () => {
    expect(wrapper.find(".loading").exists()).toBe(true);
  });
  it("p should contain the text ...", () => {
    expect(wrapper.find(".loading").text()).toBe("Loading...");
  });

  it("just testing out enzyme selectors", () => {
    expect(wrapper.find("h1 ~ p").exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
