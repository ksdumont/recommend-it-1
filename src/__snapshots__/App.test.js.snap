// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> testing matches the snapshot 1`] = `
<header
  className="app-header"
  id="header"
  title="Header"
>
  <p>
    Another Paragraph
  </p>
  <h1>
    RecommendIt
  </h1>
  <p
    className="loading"
  >
    Loading...
  </p>
</header>
`;
